packages/js/product-editor/src/blocks/product-fields/product-details-section-description/edit.tsx

```diff
@@ -0,0 +1,376 @@
+			const { isSavingEntityRecord } = select( 'core' );
+
+			return {
+				isSaving: isSavingEntityRecord< boolean >(
+					'postType',
+					'product',
+					productId
+				),
+			};
+		},
+		[ productId ]
+	);
+
+	if ( ! rootClientId ) return;
+
+	function menuItemClickHandler(
+		productTemplate: ProductTemplate,
+		onClose: () => void
+	) {
+		return async function handleMenuItemClick() {
+			try {
+				if ( ! productTemplate.layoutTemplateId ) {
+					setUnsupportedProductTemplate( productTemplate );
+					onClose();
+					return;
+				}
+
+				await validate( productTemplate.productData );
+
+				await editEntityRecord(
+					'postType',
+					'product',
+					productId,
+					productTemplate.productData
+				);
+
+				await saveEditedEntityRecord< Product >(
+					'postType',
+					'product',
+					productId,
+					{
+						throwOnError: true,
+					}
+				);
+
+				createSuccessNotice(
+					__( 'Product type changed.', 'woocommerce' )
+				);
+			} catch ( error ) {
+				const message = getProductErrorMessage( error as WPError );
+				createErrorNotice( message );
+			}
+
+			onClose();
+		};
+	}
+
+	function resolveIcon( iconId?: string | null, alt?: string ) {
+		if ( ! iconId ) return undefined;
+
+		const { Icon } = icons;
+		let icon: JSX.Element;
+
+		if ( /^https?:\/\//.test( iconId ) ) {
+			icon = <img src={ iconId } alt={ alt } />;
+		} else {
+			if ( ! ( iconId in icons ) ) return undefined;
+
+			icon = icons[ iconId as never ];
+		}
+
+		return <Icon icon={ icon } size={ 24 } />;
+	}
+
+	function getMenuItem( onClose: () => void ) {
+		return function renderMenuItem( productTemplate: ProductTemplate ) {
+			const isSelected =
+				selectedProductTemplate?.id === productTemplate.id;
+			return (
+				<MenuItem
+					key={ productTemplate.id }
+					info={ productTemplate.description ?? undefined }
+					isSelected={ isSelected }
+					icon={
+						isSelected
+							? resolveIcon( 'check' )
+							: resolveIcon(
+									productTemplate.icon,
+									productTemplate.title
+							  )
+					}
+					iconPosition="left"
+					role="menuitemradio"
+					onClick={ menuItemClickHandler( productTemplate, onClose ) }
+					className={ classNames( {
+						'components-menu-item__button--selected': isSelected,
+					} ) }
+				>
+					{ productTemplate.title }
+				</MenuItem>
+			);
+		};
+	}
+
+	async function handleModelChangeClick() {
+		try {
+			if ( isSaving ) return;
+
+			await validate( unsupportedProductTemplate?.productData );
+
+			const product = ( await saveEditedEntityRecord< Product >(
+				'postType',
+				'product',
+				productId,
+				{
+					throwOnError: true,
+				}
+			) ) ?? { id: productId };
+
+			// Avoiding to save some changes that are not supported by the current product template.
+			// So in this case those changes are saved directly to the server.
+			await saveEntityRecord(
+				'postType',
+				'product',
+				{
+					...product,
+					...unsupportedProductTemplate?.productData,
```

Comment:
`productData` property contains the unsupported type. If so, the `wc/v3/products/<product-ID>` endpoint will response with `400 Bad Request` since the value of the param type is not expected:

```
type:  "type is not one of simple, grouped, external, and variable."
```

Reply:
https://github.com/woocommerce/woocommerce/issues/46410